### PR TITLE
Update machine/host secret with full rendered ignition from MCO

### PR DESCRIPTION
**Pull request by kirankt**
This PR creates a new per machine userData secret to contain the full rendered ignition, which is used by the BaremetalOperator to build the config drive to configure the host. The original secret reference which is initially created by the openshift installer only contains pointer/stub ignition. The new secret is owned by the Machine and will be deleted when the machine is removed

co-authored-by: Steve Hardy <shardy@redhat.com>

### CODE REVIEW

File: pkg/cloud/baremetal/actuators/machine/actuator.go [hunk 1]
```diff
@@ -543,6 +552,91 @@ func (a *Actuator) setHostSpec(ctx context.Context, host *bmh.BareMetalHost, mac
...
+	}
+	if !errors.IsNotFound(err) {
+		log.Printf("Cannot get %s secret: %s", machineUserDataSecret.Name, err.Error())
+		return fmt.Errorf("Cannot get %s secret: %w", machineUserDataSecret.Name, err)
+	}
+
+	// Get the worker machineconfigpool
+	mcp := &mcfgv1.MachineConfigPool{}
+	key = client.ObjectKey{
+		Name: "worker",
```
Comment (hardys) [inline pkg/cloud/baremetal/actuators/machine/actuator.go:578]: We can't hard-code this, it'll need to be derived from the machine role to support custom machinesets

**Comment by kirankt** [comment]
RBAC PR: https://github.com/openshift/machine-api-operator/pull/766

File: pkg/cloud/baremetal/actuators/machine/actuator.go (hunk 1 above)
Comment (kirankt) [reply pkg/cloud/baremetal/actuators/machine/actuator.go]: +1. This is now fixed. Thanks for catching this one.

File: pkg/cloud/baremetal/actuators/machine/actuator.go [hunk 2]
```diff
@@ -548,6 +557,91 @@ func (a *Actuator) setHostSpec(ctx context.Context, host *bmh.BareMetalHost, mac
...
+	}
+	if !errors.IsNotFound(err) {
+		log.Printf("Cannot get %s secret: %s", machineUserDataSecret.Name, err.Error())
+		return fmt.Errorf("Cannot get %s secret: %w", machineUserDataSecret.Name, err)
+	}
+
+	// Get the worker machineconfigpool
+	mcp := &mcfgv1.MachineConfigPool{}
+	key = client.ObjectKey{
+		Name: machine.Name,
```
Comment (hardys) [inline pkg/cloud/baremetal/actuators/machine/actuator.go:583]: I don't think getting the MCP by the machine name is going to work?  The MCP is per role e.g the `machine.openshift.io/cluster-api-machine-role` label?

Comment (kirankt) [reply pkg/cloud/baremetal/actuators/machine/actuator.go]: You are correct. I've fixed this in the new update. PTAL.

**Comment by kirankt** [comment]
Depends on https://github.com/openshift/installer/pull/4456, which creates the secret asset for this PR to work correctly

**Comment by kirankt** [comment]
Obsolete. Need a different approach. MCO's rendered ignition is not complete. Additional duplication of work required to match the ignition fetched from the MachineConfigServer (MCS).
/close

**Pull request closed without merging**